packages/block-library/src/navigation-link/shared/use-entity-binding.js

```
@@ -4,6 +4,29 @@
 import { useCallback } from '@wordpress/element';
 import { useBlockBindingsUtils } from '@wordpress/block-editor';
 
+/**
+ * Builds entity binding configuration for navigation link URLs.
+ * This function generates the structure used to bind navigation link URLs to their entity sources.
+ *
+ * Using a function instead of a constant allows for future enhancements where the binding
+ * might need dynamic data (e.g., entity ID, context-specific arguments).
+ *
+ * @param {string} kind - The kind of entity ('post-type' or 'taxonomy')
+ * @return {Object} Entity binding configuration object
+ */
+export function buildNavigationLinkEntityBinding( kind ) {
+	const source = kind === 'taxonomy' ? 'core/term-data' : 'core/post-data';
```

Should we only support `post-type` or `taxonomy` here? Right now I could pass some CPT and instead of erroring it would just use the post data binding.

---

Shall we throw an error if the wrong type is used? I guess it's a severe problem. But we'd need to bail out of any entity creation as well.

---

I think we should throw an error yeah. I'm not sure how gracefully we need to handle it, since it should never happen to a user - only a developer who is messing with things.

---

Updated to throw error and NOT create the binding if error is encountered.